reader.go

```diff
@@ -247,6 +247,10 @@ func (r *Reader) ReadBytes() []byte {
 // ReadString reads a String from the Reader.
 func (r *Reader) ReadString() string {
 	b := r.readBytes("string")
+	if r.Error != nil {
```

Comment:
IMO, the fix needs to be more fundamental, in `r.Read`. `r.Read` is only ever called when an expected number of bytes should be read. Probably something like:
```
if r.head == r.tail {
  if !r.loadMore() {
    r.Error = io.ErrUnexpectedEOF
    return
  }
}
```

Reply:
sounds good, working on it

Reply:
hmm looks like the changes broke a bunch of `ocf` tests somehow

Reply:
@nrwiersma how are the files such as `testdata/full-snappy.avro` generated? I wonder if they have some invalid data in there.

Reply:
No, OCF seems to be taking advantage of the no error from before. I can take care of this in a PR if you like.

Reply:
Appreciate it! Yeah sorry was stuck on this for a bit.

Reply:
https://github.com/hamba/avro/pull/382

Reply:
thank you so much!